src/node_crypto.h

```diff
@@ -363,6 +363,12 @@ class CipherBase : public BaseObject {
     kErrorMessageSize,
     kErrorState
   };
+  enum AuthTagState {
+    kAuthTagUnknown,
+    kAuthTagLengthKnown,
```

Comment:
Is there currently a difference between these two states? Sorry for asking questions, but I’m not sure I’m fully qualified to review this…

Reply:
Yes, you can use GCM (e.g. `aes-128-gcm`) without specifying the authentication tag length, then the state will be `kAuthTagUnknown`, but other modes such as CCM (`aes-128-ccm`) and OCB (`aes-128-ocb`) require the length of the authentication tag in advance (via the `authTagLength` option), so the cipher will start in the state `kAuthTagLengthKnown`. Once the user calls `setAuthTag`, the state becomes `kAuthTagKnown`, which implies that the length is known as well.

(Never apologize for asking questions. 😉)

Reply:
Thank you! Is there any difference in how we treat those states after they have been entered?